src/ImageSharp/Image.FromStream.cs

```diff
@@ -99,6 +203,19 @@ public static IImageInfo Identify(Configuration configuration, Stream stream, ou
         public static Image Load(Stream stream, out IImageFormat format)
             => Load(Configuration.Default, stream, out format);
 
+        /// <summary>
+        /// Decode a new instance of the <see cref="Image"/> class from the given stream.
+        /// The pixel format is selected by the decoder.
+        /// </summary>
+        /// <param name="stream">The stream containing image information.</param>
+        /// <exception cref="ArgumentNullException">The stream is null.</exception>
+        /// <exception cref="NotSupportedException">The stream is not readable.</exception>
+        /// <exception cref="UnknownImageFormatException">Image format not recognised.</exception>
+        /// <exception cref="InvalidImageContentException">Image contains invalid content.</exception>
+        /// <returns>A <see cref="Task{FormattedImage}"/> representing the asynchronous operation.</returns>
+        public static Task<FormattedImage> LoadWithFormatAsync(Stream stream)
```

Comment:
Wondering whether it would be better to move this family of methods under `FormattedImage` and `FormattedImageInfo`. Cleaner naming, but bad for discoverability.

Reply:
Structurally cleaner yeah, but I think we'll end up confusing people if we split out the async API. I'd rather avoid the questions.